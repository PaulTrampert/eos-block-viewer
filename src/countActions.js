export default (block) => {
  let actions = 0;
  for (let transaction of block.transactions) {
    if (transaction.trx.transaction) {
      actions += transaction.trx.transaction.actions.length;
    }
  }
  return actions;
}