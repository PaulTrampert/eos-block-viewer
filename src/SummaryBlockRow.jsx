import React from 'react';
import PropTypes from 'prop-types';
import countActions from './countActions';

const SummaryBlockRow = ({block, onClick}) => (
  <tr onClick={onClick}>
    <td>{block.id}</td>
    <td>{new Date(block.timestamp).toLocaleString()}</td>
    <td>{countActions(block)}</td>
  </tr>
);

SummaryBlockRow.propTypes = {
  block: PropTypes.object.isRequired,
  onClick: PropTypes.func.isRequired
};

export default SummaryBlockRow;