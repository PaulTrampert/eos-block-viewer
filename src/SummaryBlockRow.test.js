import React from 'react';
import {shallow} from 'enzyme';
import SummaryBlockRow from './SummaryBlockRow';
import countActions from './countActions';

jest.mock('./countActions');
jest.spyOn(global, 'Date')
  .mockImplementation(() => ({
    toLocaleString: () => "this is the local representation of datetime"
  }));

describe('SummaryBlockRow', () => {
  let subject;
  let block;
  let onClick;

  beforeEach(() => {
    block = {
      id: 'asdfasdf',
      timestamp: '2020-02-08T20:20:20',
      transactions: []
    };

    onClick = jest.fn();

    countActions.mockReturnValue(0);

    subject = shallow(<SummaryBlockRow block={block} onClick={onClick} />);
  });

  describe('render', () => {
    it('binds the onClick event', () => {
      expect(subject.find('tr').props().onClick).toBe(onClick);
    });

    it('counts the number of actions', () => {
      expect(countActions).toHaveBeenCalledWith(block);
    });

    it("renders the block's id, timestamp, and the number of actions", () => {
      expect(subject).toMatchSnapshot();
    });
  });
});