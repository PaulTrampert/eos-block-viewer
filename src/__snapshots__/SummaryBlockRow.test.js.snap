// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryBlockRow render renders the block's id, timestamp, and the number of actions 1`] = `
<tr
  onClick={[MockFunction]}
>
  <td>
    asdfasdf
  </td>
  <td>
    2/8/2020, 3:20:20 PM
  </td>
  <td>
    0
  </td>
</tr>
`;
